src/navigation-plan.js

```diff
@@ -11,6 +12,15 @@ export function buildNavigationPlan(navigationContext, forceLifecycleMinimum) {
   var next = navigationContext.nextInstruction;
   var plan = {}, viewPortName;
 
+  if ('redirect' in next.config) {
+    let redirectLocation = resolveUrl(next.config.redirect, getInstructionBaseUrl(next));
+    if (next.queryString) {
+      redirectLocation += '?' + next.queryString;
+    }
+
+    return Promise.reject(new Redirect(redirectLocation));
```

Comment:
We need to do this in `buildNavigationPlan` because a redirect could be configured in any node of the tree. The redirect must be triggered as soon as we find one to avoid trying to build a plan for a navigation that won't happen (a module ID is needed for that, which we don't have).